### DISCUSSION TITLE
How to skip ci with specific labels?

### DISCUSSION THREAD

**Discussion by littleGnAl** [cirruslabs/cirrus-ci-docs#1168] [Q&A]
I'm trying to implement the skip ci feature by filtering the label with "ci:skip", my config is here
https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK/blob/47595f9a4cd4c065c10418fb3236e410bd73e852/.cirrus.yml#L7

But it seems not to work correctly, any suggestions for it?

**Comment by fkorotkov** [comment]
Hey @littleGnAl,

It seems you need to modify your matching expression to `.*ci:skip.*`:

```
$CIRRUS_PR_LABELS !=~ '.*ci:skip.*'
```

**Comment by littleGnAl** [comment]
@fkorotkov I had tried it before, but it's not working. So I change it to `$CIRRUS_PR_LABELS !=~ 'ci:skip'`, and it work for a while after I change it, but it's not worked recently.
https://github.com/AgoraIO-Extensions/iris_method_channel_flutter/commit/7fdc6e51995f1c9bf153eebc9e8a61bd679f1648

**Comment by fkorotkov** [comment]
I think in your case the error is in `$CIRRUS_PR != ''` part. This makes you only run Cirrus for PRs. Link you provided is for a `main` branch build. It seems you can actually just use `skip` functionality with the following expression:

```yaml
task:
  skip: $CIRRUS_REPO_OWNER != 'AgoraIO-Extensions' || $CIRRUS_PR_LABELS =~ '.*ci:skip.*'
```

**Reply by littleGnAl** [reply]
But I want to run the build in PR only.

**Reply by fkorotkov** [reply]
Then the linked check suite behaved as expected. `$CIRRUS_PR` was empty and it didn't run anything.

**Comment by littleGnAl** [comment]
I make a change following your suggestion, but it seems not to work.
Here's a fix PR https://github.com/AgoraIO-Extensions/iris_method_channel_flutter/pull/69, it works if I add the lablel "ci:skip", but after the PR merged, I open another PR with the label "ci:skip", it seems the `skip` not work https://github.com/AgoraIO-Extensions/iris_method_channel_flutter/pull/70.
